src/libostree/ostree-sign-ed25519.c

```diff
@@ -415,17 +415,18 @@ ostree_sign_ed25519_add_pk (OstreeSign *self, GVariant *public_key, GError **err
   if (!_ostree_sign_ed25519_is_initialized (sign, error))
     return FALSE;
 
-  gpointer key = NULL;
+  g_autofree guint8 *key_owned = NULL;
+  const guint8 *key = NULL;
```

Comment:
Why not:

`g_autofree const guint8 *key = NULL;`

key_owned is unused, just set.

Reply:
Because `g_variant_get_fixed_array()` returns a `const` pointer into its data, whereas `g_base64_decode` returns a new value.  Notice that we do set `key_owned` only in the latter case.

Reply:
Ah I see, so you want it free'd at the end of scope in one case but not the other... Makes sense